src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -137,6 +137,10 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
                 return _lastKnownLocation;
             }
 
+            if (_requestMethod == RequestMethod.Patch)
```

Comment:
This condition isn't mutually exclusive with the previous one. Is it possible that `_requestMethod == RequestMethod.Patch`, but previous condition is also true?

Reply:
Yeah, thanks for your reminder and I think I find an example. When trying to update the profile in cdn, I got this original response header.
```
"ResponseHeaders": {
        "Azure-AsyncOperation": "https://management.azure.com/subscriptions/mySubscriptions/resourcegroups/testRg-8080/providers/Microsoft.Cdn/operationresults/9a06c2d3-1ace-43e6-8839-126f347e6ba7?api-version=2020-09-01",
        "Cache-Control": "no-cache",
        "Content-Length": "354",
        "Content-Type": "application/json; charset=utf-8",
        "Date": "Thu, 16 Sep 2021 09:16:36 GMT",
        "Expires": "-1",
        "Location": "https://management.azure.com/subscriptions/mySubscriptions/resourcegroups/testRg-8080/providers/Microsoft.Cdn/operationresults/9a06c2d3-1ace-43e6-8839-126f347e6ba7/profileresults/profile-9026?api-version=2020-09-01",
        "Pragma": "no-cache",
        "Retry-After": "10",
        "Server": "Kestrel",
        "Strict-Transport-Security": "max-age=31536000; includeSubDomains",
        "X-Content-Type-Options": "nosniff",
        "x-ms-client-request-id": "96c6552e4b02bd98e985cef1b242b4c3",
        "x-ms-correlation-request-id": "9a8d4285-a893-4d6d-9f32-351ba6088576",
        "x-ms-ratelimit-remaining-subscription-resource-requests": "23",
        "x-ms-request-id": "36d6d7be-a075-42e7-bfee-15e80f81dc94",
        "x-ms-routing-request-id": "SOUTHEASTASIA:20210916T091637Z:9a8d4285-a893-4d6d-9f32-351ba6088576"
      }
```
In this patch mehod, the _originalResponseHasLocation will be true and _finalStateVia will be Location, so the previous condition will be met and _lastKnownLocation will be treated as FinalUri, which will also cause the construct resource to fail.
Therefore, I think a more effective change would be move the judgement of patch up：
```
if (_requestMethod == RequestMethod.Put || _requestMethod == RequestMethod.Patch || _originalResponseHasLocation && _finalStateVia == OperationFinalStateVia.OriginalUri)
{
    return _startRequestUri.AbsoluteUri;
}
```

Reply:
Ok, can you please add a test for it as well?

Reply:
> 

Sure, I'm working on the test.

Reply:
consider to add comments, otherwise it looks some magic happened...